include/boundary_standard.hxx

```diff
@@ -10,107 +10,91 @@
 
 #include <utility>
 
-/// Dirichlet boundary condition set half way between guard cell and grid cell at 2nd order accuracy
-class BoundaryDirichlet_2ndOrder : public BoundaryOp {
- public:
-  BoundaryDirichlet_2ndOrder() : val(0.) {}
-  BoundaryDirichlet_2ndOrder(BoutReal setval ): val(setval) {}
-  BoundaryDirichlet_2ndOrder(BoundaryRegion *region, BoutReal setval=0.):BoundaryOp(region),val(setval) { }
-  BoundaryOp* clone(BoundaryRegion *region, const list<string> &args) override;
-
-  using BoundaryOp::apply;
-  void apply(Field2D &f) override;
-  void apply(Field3D &f) override;
-
-  using BoundaryOp::apply_ddt;
-  void apply_ddt(Field2D &f) override;
-  void apply_ddt(Field3D &f) override;
- private:
-  BoutReal val;
-};
-
 /// Dirichlet (set to zero) boundary condition
 class BoundaryDirichlet : public BoundaryOp {
  public:
-  BoundaryDirichlet() : gen(nullptr) {}
-  BoundaryDirichlet(BoundaryRegion *region, std::shared_ptr<FieldGenerator> g)
-      : BoundaryOp(region), gen(std::move(g)) {}
+  using BoundaryOp::BoundaryOp; // inherit BoundaryOp constructors
   BoundaryOp* clone(BoundaryRegion *region, const list<string> &args) override;
 
   using BoundaryOp::apply;
-  void apply(Field2D &f) override;
-  void apply(Field2D &f,BoutReal t) override;
-  void apply(Field3D &f) override;
-  void apply(Field3D &f,BoutReal t) override;
+  void apply(Field2D &f,BoutReal t = 0.) override {
+    applyTemplate<Field2D>(f, t);
```

Comment:
You probably don't actually need to specify `Field2D` or `Field3D` here as the compiler should be able to infer the type, although being explicit certainly doesn't hurt.